src/ng/filter/orderBy.js

```diff
@@ -86,6 +86,8 @@
  *
  * **Note:** If you notice numbers not being sorted as expected, make sure they are actually being
  *           saved as numbers and not strings.
+ * **Note:** If null values are not being sorted as expected it may be due to rule number
+ *           1 above which sorts `null` as the string `'null'`.
```

Comment:
No, it is not related to rule number 1. Normally, the type of `null` is `object`. But we sort it as if it's type were `string`.

I would write something like:

> **Note:** Normally the type of `null` is `object`. When applying the above algorithm, `null` values are converted to the string `'null'` (and thus have a type of `string`).